subsys/net/l2/wifi/wifi_utils.c

```diff
@@ -241,8 +241,7 @@ int wifi_utils_parse_scan_bands(char *scan_bands_str, uint8_t *band_map)
 		return -EINVAL;
 	}
 
-	strncpy(parse_str, scan_bands_str, len);
-	parse_str[len] = '\0';
+	strncpy(parse_str, scan_bands_str, sizeof(parse_str));
```

Comment:
The warning is false negative, we do allocate space for `NULL` and also do a length check before copying, so, it can never truncate.

Reply:
Indeed, the original code is working just fine. But if it makes gcc happy the change could be accepted too.
Could you add the gcc version that is emitting the warning to the commit message for a reference.

Reply:
> The warning is false negative, we do allocate space for NULL and also do a length check before copying, so, it can never truncate.

Sure- I'm aware, just trying to please gcc :). I'll add the gcc version to the commit message